### PR TITLE
Allow users to set notifications as read without a message id

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner @leogrim 

This is a quick fix so that you can mark entire user threads as read/unread even if there are no messages on it. This loses some granularity but ensures that you can mark an initial send as read (an outstanding bug).

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala [hunk 1]
```diff
@@ -185,22 +185,24 @@ class MessageThreadNotificationBuilderImpl @Inject() (
...
       case (keepId, thread) =>
         val thread = threadsById(keepId)
+        val userThreadOpt = userThreadById.get(keepId)
         val messageOpt = lastMsgById.get(keepId)
         val threadStarter = basicUserByIdMap(thread.startedBy).externalId
         val threadActivity = threadActivityById(keepId)
         val MessageCount(numMessages, numUnread) = msgCountById(keepId)
-        val muted = mutedById(keepId)
+        val muted = userThreadOpt.exists(_.muted)
+        val unread = userThreadOpt.map(_.unread).getOrElse(!messageOpt.exists(_.from.asUser.safely.contains(userId)))
```
Comment (leogrim) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala:205]: `userThreadOpt.exists(_.unread) || !messageOpt.exists(_.from.asUser.safely.contains(userId))`

File: kifi-backend/modules/shoebox/marketing [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 27c110d891ab32e7912deb07ae8f9d031f627f03
+Subproject commit cd97c7c0075f4902df6b698ec5dfa640c15591a4
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/marketing:1]: ?

**Comment by leogrim** [comment]
cool! :shipit:

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala:205]: I want to use the value of `unread` if the user thread exists, whether it's true or false.

File: packrat/main.js [hunk 3]
```diff
@@ -1204,12 +1204,12 @@ api.port.on({
...
     whenTabFocused(tab, o.keepId, function (tab) {
       markRead(o.keepId, o.eventId, o.time);
-      socket.send(['set_message_read', o.eventId]);
+      socket.send(['set_message_read', o.eventId, o.keepId]);
     });
   },
   set_message_read: function (o, _, tab) {
     markRead(o.threadId, o.messageId, o.time);
-    socket.send(['set_message_read', o.messageId]);
+    socket.send(['set_message_read', o.messageId, o.threadId]);
```
Comment (camhashemi) [inline packrat/main.js:1212]: @cvializ we need to deploy the ext to fix this bug (can't mark messages from mobile as read). I forgot how to :/

**Pull request merged**